analyzers/its/expected/BlazorSample/BlazorSample--net7.0-S1451.json

```diff
@@ -43,6 +43,32 @@
 "id":  "S1451",
 "message":  "Add or update the header of this file.",
 "location":  {
+"uri":  "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/sources/BlazorSample/BlazorSample/Pages/S6798/S6798.CSharpOnly.cs#L1",
```

Comment:
This one should include also razor files, I think.
However, it may not be feasible, if the header is written as HTML comment (`<!-- -->`)  or as a Razor comment (`@**@`). Should we have a rule improvement issue?

Reply:
If it's a html component I don't think we can, not sure for the razor comment. Anyway, I think most people will set the header with the options you mentioned so one non-tracked option is enough to raise a load of FPs. For this reason, I would not extend the rule to razor files. Happy to discuss

Reply:
I don't think we can do it, as both types of comments are not reported to the generated files, we have no way to see if there is a header for the file.
So until we can access Razor SyntaxTree I would not include `.razor` files.

Reply:
I agree that fixing this false negative may not be feasible with our current approach if we focus at the AST level.
I would however consider the false negative as a valid one: `razor` files, as well as `cshtml` files are to be considered "production source" like `cs` files. If the policy is to add a disclaimer to every piece of production source, S1451 should include `razor` and `cshtml` files.

Moreover, the rule is unfeasible only if we restrain our analysis to the AST provided by Roslyn. If we find out that Roslyn-Razor integration may take longer than expected, and that we want to deliver rules on pure markup as we find quite valuable for Razor/Blazor users, we may start to analyze sections of markup from source text, rather than solely based on Roslyn AST.

For these reasons, I have created an issue in the backlog, and explained that:
- there are technical difficulties to be overcome
- we may want to wait for a tighter Roslyn-Razor integration
- `cshtml` and `razor` need to be treated differently